File: Anul II/Medii de proiectare si programare/Laborator/lab02/Java/src/main/java/com/ubb/model/RaceEntry.java
package com.ubb.model;

import java.util.Objects;

public class RaceEntry extends Identifiable<Long> {
    private Participant participant;
    private Race race;

    public RaceEntry(Long ID, Participant participant, Race race) {
        setID(ID);
        this.participant = participant;
        this.race = race;
    }

    public Participant getParticipant() {
        return participant;
    }

    public Race getRace() {
        return race;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        RaceEntry raceEntry = (RaceEntry) o;
        return getParticipant().equals(raceEntry.getParticipant()) && getRace().equals(raceEntry.getRace());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getParticipant(), getRace());
    }
}

File: Anul II/Medii de proiectare si programare/Laborator/lab08/C#/Ubb.BikeContest.Model/src/RaceEntry.cs
namespace Ubb.BikeContest.Model;

[Serializable]
public class RaceEntry : Identifiable<long>
{
    private readonly Race _race;
    private readonly Participant _participant;

    public RaceEntry(Participant participant, Race race)
    {
        _participant = participant;
        _race = race;
    }

    public Participant Participant => _participant;

    public Race Race => _race;
}